### PR TITLE
VideoCapture timeout set/get

### PR DESCRIPTION
**Pull request by iyadahmed**
The original PR is amazing, a fixed timeout is not that convenient however, so this small patch adds get and set option for timeout
https://github.com/opencv/opencv/pull/6053

### CODE REVIEW

File: modules/videoio/include/opencv2/videoio.hpp [hunk 1]
```diff
@@ -179,6 +179,8 @@ enum VideoCaptureProperties {
        CAP_PROP_BITRATE       =47, //!< (read-only) Video bitrate in kbits/s
        CAP_PROP_ORIENTATION_META=48, //!< (read-only) Frame rotation defined by stream meta (applicable for FFmpeg back-end only)
        CAP_PROP_ORIENTATION_AUTO=49, //!< if true - rotates output frames of CvCapture considering video file's metadata  (applicable for FFmpeg back-end only) (https://github.com/opencv/opencv/issues/15499)
+       CAP_PROP_OPEN_TIMEOUT_MSEC  =53, //!< (**open-only**) timeout in milliseconds for opening a video capture (applicable for FFmpeg back-end only)
+       CAP_PROP_READ_TIMEOUT_MSEC  =54, //!< timeout in milliseconds for reading from a video capture (applicable for FFmpeg back-end only)
```
Comment (asmorkalov) [inline modules/videoio/include/opencv2/videoio.hpp:183]: Please follow common formatting approach. No need 2 spaces pefore equal sign.

**Review by asmorkalov — COMMENTED** [review]
Thanks a lot for the patch! Looks good in general.

**Comment by iyadahmed** [comment]
> Thanks a lot for the patch! Looks good in general.

You are welcome 🙏
Just needs some work I think for some reason build fails on the cloud server

File: modules/videoio/include/opencv2/videoio.hpp (hunk 1 above)
Comment (iyadahmed) [reply modules/videoio/include/opencv2/videoio.hpp:183]: ok will do so

**Comment by iyadahmed** [comment]
hmm not sure why the checks fail

**Comment by alalek** [comment]
AFAIK, `cap_ffmpeg_impl.hpp` doesn't have access to enums from `videoio.hpp`

Please use enum values aligned with the "master" branch (=53 is free there), so we the same values between OpenCV versions.

**Comment by iyadahmed** [comment]
> AFAIK, `cap_ffmpeg_impl.hpp` doesn't have access to enums from `videoio.hpp`
> 
> ~Please use enum values aligned with the "master" branch (=53 is free there), so we the same values between OpenCV versions.~ (all is done here)

ah ok

**Comment by alalek** [comment]
If you don't need this feature for OpenCV 3.x release series, then it makes sense to retarget this PR to "master" (4.x) branch. There is no such problem.

**Comment by iyadahmed** [comment]
> If you don't need this feature for OpenCV 3.x release series, then it makes sense to retarget this PR to "master" (4.x) branch. There is no such problem.

ah no worries, I need it for both, will do another pull after fixing the small issue

**Comment by iyadahmed** [comment]
Ok this should fix it, was working on master, 3.4 had FFMPEG enums in different place

**Comment by iyadahmed** [comment]
Checks passed

File: modules/videoio/src/cap_ffmpeg_api.hpp [hunk 2]
```diff
@@ -30,7 +30,9 @@ enum
     CV_FFMPEG_CAP_PROP_CODEC_PIXEL_FORMAT=46,
     CV_FFMPEG_CAP_PROP_BITRATE=47,
     CV_FFMPEG_CAP_PROP_ORIENTATION_META=48,
-    CV_FFMPEG_CAP_PROP_ORIENTATION_AUTO=49
+    CV_FFMPEG_CAP_PROP_ORIENTATION_AUTO=49,
+    CV_FFMPEG_CAP_PROP_OPEN_TIMEOUT_MSEC=50,
+    CV_FFMPEG_CAP_PROP_READ_TIMEOUT_MSEC=51
```
Comment (alalek) [inline modules/videoio/src/cap_ffmpeg_api.hpp:35]: Need to restore defined enum values in videoio.hpp. It is a public API header, from which users know how to use this feature.

Also please use 53, 54 as values to align with master (for both "enums").

Comment (iyadahmed) [reply modules/videoio/src/cap_ffmpeg_api.hpp:35]: ok will do now

Comment (iyadahmed) [reply modules/videoio/src/cap_ffmpeg_api.hpp]: Done

**Review by alalek — APPROVED** [review]
Well done! Thank you for contribution :+1:

**Comment by iyadahmed** [comment]
Btw I should make another PR for master right?

**Comment by iyadahmed** [comment]
> It would be nice, because there are several differences.
> 
> On master branch we have [`VideoCaptureParameters`](https://github.com/opencv/opencv/blame/4.5.3/modules/videoio/src/cap_ffmpeg_impl.hpp#L876) which can pass "open timeout" properly. On 3.4 it doesn't work as expected ("set" property is always called after "open").

I see, will do a PR asap

**Pull request merged**